src/components/QueryEditor/PPLFormatEditor/index.test.tsx

```diff
@@ -1,39 +1,51 @@
 import React from 'react';
-import { shallow } from 'enzyme';
 import { PPLFormatEditor } from './';
-import { QueryEditorRow } from '../QueryEditorRow';
-import { SettingsEditor } from './SettingsEditor';
-import { OpenCloseButton } from './OpenCloseButton';
-import { HelpMessage } from './HelpMessage';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { OpenSearchQuery, QueryType } from 'types';
+import { OpenSearchDatasource } from 'datasource';
+import { OpenSearchProvider } from '../OpenSearchQueryContext';
 
-jest.mock('../OpenSearchQueryContext', () => ({
-  useQuery: jest.fn(() => ({
-    format: 'time_series',
-  })),
-}));
+const pplLogsQuery: OpenSearchQuery = {
+  refId: 'A',
+  queryType: QueryType.PPL,
+  format: 'time_series',
+  query: 'source = test-index',
+  bucketAggs: [{ type: 'date_histogram', id: '2' }],
+  metrics: [{ id: '1', type: 'count' }],
+};
 
+const setup = () => {
+  render(
+    <OpenSearchProvider
+      query={pplLogsQuery}
+      datasource={{} as OpenSearchDatasource}
+      onChange={jest.fn()}
+    >
+      <PPLFormatEditor />
+    </OpenSearchProvider>
+  );
+};
 describe('PPLFormatEditor', () => {
   it('should render correctly', () => {
-    shallow(<PPLFormatEditor />);
+    setup();
   });
 
-  it('should render all components of PPL format editor row', () => {
-    const wrapper = shallow(<PPLFormatEditor />);
-    const queryEditorRow = wrapper.find(QueryEditorRow);
-    expect(queryEditorRow.length).toBe(1);
-    expect(queryEditorRow.props().label).toBe('Format');
-    const settingsEditor = wrapper.find(SettingsEditor);
-    expect(settingsEditor.props().value).toBe('time_series');
-    expect(wrapper.find(OpenCloseButton).length).toBe(1);
-    expect(wrapper.find(HelpMessage).length).toBe(0);
+  it('should render all components of PPL format editor row', async () => {
+    setup();
+    await waitFor(() => {
+      expect(screen.getByText('Format')).toBeInTheDocument();
+      expect(screen.getByText('Time series')).toBeInTheDocument();
+      expect(screen.getByText('Show help')).toBeInTheDocument();
+      expect(screen.queryByTestId('help-message')).not.toBeInTheDocument();
+    });
   });
-
-  it('should show help message on click', () => {
-    const wrapper = shallow(<PPLFormatEditor />);
-    wrapper
-      .find(OpenCloseButton)
-      .props()
-      .onClick();
-    expect(wrapper.find(HelpMessage).length).toBe(1);
+  it('should show help message on click', async () => {
+    setup();
+    await waitFor(async () => {
+      const button = screen.getByText('Show help');
+      await userEvent.click(button);
```

Comment:
WaitFor is a tricky beast, I'd recommend checking out https://kentcdodds.com/blog/common-mistakes-with-react-testing-library there are a few instances in this pr that I think might be worth taking a look at how we're using it.

For example in this case, if I understand correctly waitFor will actually call multiple times, so in this case you're basically clicking the button multiple times. 

Might be good to see is it possible to not use waitfor here? If not figure out why, what exactly are we waiting for? What in the dom can we use as a marker to say "hey the thing we are waiting for is done". Do we need to wait for all of these things for example? Or could we wait for the button to show the correct text and then continue? 

Could we use find instead of waitfor, etc.

Reply:
Didn't know this about the waitFor callback! The reason I'm waiting for the setup is that there is some useContext stuff that needs to happen for the component to render, but I can certainly use find, that would also make a lot of sense.